src/azure-cli/azure/cli/command_modules/deploymentmanager/_params.py

```diff
@@ -119,24 +119,27 @@ def load_arguments(self, _):
 
     duration_type = CLIArgumentType(options_list='--duration', help='The duration of the wait step in ISO 8601 format.')
     step_name_type = CLIArgumentType(options_list=['--step-name', '--name', '-n'], help='The name of the step', completer=get_resource_name_completion_list('Microsoft.DeploymentManager/steps'))
+    step_type = CLIArgumentType(options_list='--step', help='The step object')
```

Comment:
does --step-type only work for json file? what about json string? how can users know which kind of json is valid?

Reply:
It takes either a string or a file. Updated the help text.

Reply:
JSON is a little complicated for users to input in command line and it is hard to show up allowed vaules/type for each parameter. Can we unfold the json object for easy usage or may I know why you want to use json here? For more information, please refer to https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#commands-with-complex-types

Reply:
Check how the sample JSON looks like in this section here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-manager-health-check#how-service-health-is-determined

There are multiple types of collections and arguments in the 'properties' object in the above JSON are supposed to be provided on the command line, which is cumbersome. If the users have multiple health checks as part of a step, which is very typical, then this leads to an array of such different collections, if we were to model this as numerous commands. Hence, we modeled it to be easy for users to copy a model from samples, modify and use the same. It is consistent with our ARM template schema.